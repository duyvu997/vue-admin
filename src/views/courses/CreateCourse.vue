<template>
  <div class="form-create-course-container">
    <el-form ref="form" :model="form" :rules="rules" label-width="160px">
      <el-form-item label="Tên khoá học" prop="name">
        <el-input v-model="form.name" />
      </el-form-item>
      <el-form-item label="Thời lượng khoá học" prop="duration">
        <div class="duration-field">
          <div class="field" style="margin-right: 10px">
            <el-input v-model="form.duration" />
          </div>
          <div class="field">
            <el-select
              v-model="form.durationType"
              placeholder="Lựa chọn đơn vị thời gian"
              style="width: 100%;"
            >
              <el-option label="giây" value="giây" />
              <el-option label="phút" value="phút" />
            </el-select>
          </div>
        </div>
      </el-form-item>
      <el-form-item label="Thời gian học" prop="date">
        <el-date-picker
          v-model="form.date"
          type="date"
          placeholder="Lựa chọn thời gian học"
          style="width: 100%;"
        />
      </el-form-item>
      <el-form-item label="Thông điệp ngắn" prop="shortMessage">
        <el-input v-model="form.shortMessage" />
      </el-form-item>
      <el-form-item label="Mô tả" prop="description">
        <el-input v-model="form.description" type="textarea" />
      </el-form-item>
      <el-form-item label="Trạng thái" prop="status">
        <el-switch v-model="form.status" />
      </el-form-item>
      <el-form-item label="Độ tuổi tối thiểu" prop="minimumAge">
        <el-input v-model="form.minimumAge" />
      </el-form-item>
      <el-form-item label="Độ tuổi tối đa" prop="maximumAge">
        <el-input v-model="form.maximumAge" />
      </el-form-item>
      <el-form-item label="Hình ảnh" prop="picture">
        <el-upload
          :file-list="form.picture"
          :limit="1"
          :auto-upload="false"
          action="https://test.io"
          style="display: flex"
        >
          <el-button style="margin-right: 10px;">
            Chọn hình ảnh
          </el-button>
        </el-upload>
      </el-form-item>
      <el-form-item>
        <el-button @click="onCancel">
          Huỷ
        </el-button>
        <el-button type="primary" @click="onSubmit('form')">
          Tạo
        </el-button>
      </el-form-item>
    </el-form>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import store from '@/store/index'
import { courseAction, CREATE_COURSE } from '@/store/course/action.type'

export default Vue.extend({
  name: 'CreateCourse',
  data() {
    return {
      form: {
        name: '',
        duration: '',
        durationType: 'giây',
        date: '',
        shortMessage: '',
        description: '',
        status: false,
        minimumAge: '',
        maximumAge: '',
        picture: []
      },
      rules: {
        name: [
          { required: true, message: 'Không được để trống', trigger: 'change' }
        ],
        duration: [
          { required: true, message: 'Không được để trống', trigger: 'change' }
        ],
        date: [
          { required: true, message: 'Không được để trống', trigger: 'change' }
        ],
        description: [
          { max: 5000, message: 'Tối đa 5000 kí tự', trigger: 'change' }
        ],
        minimumAge: [
          { required: true, message: 'Không được để trống', trigger: 'change' }
        ],
        maximumAge: [
          { required: true, message: 'Không được để trống', trigger: 'change' }
        ]
      }
    }
  },
  methods: {
    /* istanbul ignore next */
    formRef(
      formName: string
    ): Vue & { validate: (func: (valid: boolean) => boolean) => void } {
      return this.$refs[formName] as Vue & {
        validate: (func: (valid: boolean) => boolean) => void
      }
    },
    /* istanbul ignore next */
    onSubmit(formName: string) {
      this.formRef(formName).validate((valid: boolean) => {
        if (valid) {
          const courseTobeCreated = {
            description: this.form.description,
            duration: this.form.duration,
            endDate: '',
            fileId: null,
            image: '',
            minAge: this.form.minimumAge,
            name: this.form.name,
            shortMessage: this.form.shortMessage,
            startDate: this.form.date,
            status: this.form.status ? 'ACTIVE' : 'DEACTIVE'
          }
          store.dispatch(courseAction(CREATE_COURSE), courseTobeCreated)
          alert('submit!')
          return true
        } else {
          console.log('error submit!!')
          return false
        }
      })
    },
    /* istanbul ignore next */
    onCancel() {
      this.$router.push({
        path: '/courses/index'
      })
    }
  }
})
</script>

<style lang="scss" scoped>
.form-create-course {
  &-container {
    margin: 30px;

    .duration-field {
      display: flex;

      .field {
        flex: 1 0;
      }
    }
  }
}
</style>
