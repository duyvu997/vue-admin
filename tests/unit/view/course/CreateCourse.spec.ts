import { mount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import ElementUI from "element-ui";
import CreateCourse from "@/views/courses/CreateCourse.vue";

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(ElementUI);

const routes = [
  {
    path: "/courses",
    children: [
      {
        path: "create"
      }
    ]
  }
];

const router = new VueRouter({
  routes
});

const wrapper = mount(CreateCourse, {
  localVue,
  router
});

describe("CreateCourse.vue", () => {
  it("snapshot testing", async () => {
    router.push("/courses/create");
    await wrapper.vm.$nextTick();
    expect(wrapper.element).toMatchSnapshot() 
 });
});
