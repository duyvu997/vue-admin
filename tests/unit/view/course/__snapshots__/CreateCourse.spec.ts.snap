// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCourse.vue snapshot testing 1`] = `
<div
  class="form-create-course-container"
>
  <form
    class="el-form"
  >
    <div
      class="el-form-item is-required"
    >
      <label
        class="el-form-item__label"
        for="name"
        style="width: 160px;"
      >
        Tên khoá học
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-input"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            type="text"
          />
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item is-required"
    >
      <label
        class="el-form-item__label"
        for="duration"
        style="width: 160px;"
      >
        Thời lượng khoá học
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="duration-field"
        >
          <div
            class="field"
            style="margin-right: 10px;"
          >
            <div
              class="el-input"
            >
              <!---->
              <input
                autocomplete="off"
                class="el-input__inner"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <div
            class="field"
          >
            <div
              class="el-select"
              style="width: 100%;"
            >
              <!---->
              <div
                class="el-input el-input--suffix"
              >
                <!---->
                <input
                  autocomplete="off"
                  class="el-input__inner"
                  placeholder="Lựa chọn đơn vị thời gian"
                  readonly="readonly"
                  type="text"
                />
                <!---->
                <span
                  class="el-input__suffix"
                >
                  <span
                    class="el-input__suffix-inner"
                  >
                    <i
                      class="el-select__caret el-input__icon el-icon-arrow-up"
                    />
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                    <!---->
                  </span>
                  <!---->
                </span>
                <!---->
                <!---->
              </div>
              <transition-stub
                name="el-zoom-in-top"
              >
                <div
                  class="el-select-dropdown el-popper"
                  style="display: none;"
                >
                  <div
                    class="el-scrollbar"
                    style=""
                  >
                    <div
                      class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default"
                    >
                      <ul
                        class="el-scrollbar__view el-select-dropdown__list"
                      >
                        <!---->
                        <li
                          class="el-select-dropdown__item selected"
                        >
                          <span>
                            giây
                          </span>
                        </li>
                         
                        <li
                          class="el-select-dropdown__item"
                        >
                          <span>
                            phút
                          </span>
                        </li>
                      </ul>
                    </div>
                    <div
                      class="el-scrollbar__bar is-horizontal"
                    >
                      <div
                        class="el-scrollbar__thumb"
                        style="transform: translateX(0%); webkit-transform: translateX(0%);"
                      />
                    </div>
                    <div
                      class="el-scrollbar__bar is-vertical"
                    >
                      <div
                        class="el-scrollbar__thumb"
                        style="transform: translateY(0%); webkit-transform: translateY(0%);"
                      />
                    </div>
                  </div>
                  <!---->
                </div>
              </transition-stub>
            </div>
          </div>
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item is-required"
    >
      <label
        class="el-form-item__label"
        for="date"
        style="width: 160px;"
      >
        Thời gian học
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date"
          style="width: 100%;"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            name=""
            placeholder="Lựa chọn thời gian học"
            type="text"
          />
          <span
            class="el-input__prefix"
          >
            <i
              class="el-input__icon el-icon-date"
            />
            <!---->
          </span>
          <span
            class="el-input__suffix"
          >
            <span
              class="el-input__suffix-inner"
            >
              <i
                class="el-input__icon"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </span>
            <!---->
          </span>
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item"
    >
      <label
        class="el-form-item__label"
        for="shortMessage"
        style="width: 160px;"
      >
        Thông điệp ngắn
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-input"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            type="text"
          />
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item"
    >
      <label
        class="el-form-item__label"
        for="description"
        style="width: 160px;"
      >
        Mô tả
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-textarea"
        >
          <textarea
            autocomplete="off"
            class="el-textarea__inner"
            style="min-height: -4px;"
          />
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item"
    >
      <label
        class="el-form-item__label"
        for="status"
        style="width: 160px;"
      >
        Trạng thái
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-switch"
          role="switch"
        >
          <input
            class="el-switch__input"
            name=""
            true-value="true"
            type="checkbox"
          />
          <!---->
          <span
            class="el-switch__core"
            style="width: 40px;"
          />
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item is-required"
    >
      <label
        class="el-form-item__label"
        for="minimumAge"
        style="width: 160px;"
      >
        Độ tuổi tối thiểu
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-input"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            type="text"
          />
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item is-required"
    >
      <label
        class="el-form-item__label"
        for="maximumAge"
        style="width: 160px;"
      >
        Độ tuổi tối đa
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          class="el-input"
        >
          <!---->
          <input
            autocomplete="off"
            class="el-input__inner"
            type="text"
          />
          <!---->
          <!---->
          <!---->
          <!---->
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item"
    >
      <label
        class="el-form-item__label"
        for="picture"
        style="width: 160px;"
      >
        Hình ảnh
      </label>
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <div
          style="display: flex;"
        >
          <div
            class="el-upload el-upload--text"
            tabindex="0"
          >
            <button
              class="el-button el-button--default"
              style="margin-right: 10px;"
              type="button"
            >
              <!---->
              <!---->
              <span>
                
          Chọn hình ảnh
        
              </span>
            </button>
            <input
              class="el-upload__input"
              name="file"
              type="file"
            />
          </div>
          <transition-group-stub
            class="el-upload-list el-upload-list--text"
            name="el-list"
            tag="ul"
          />
        </div>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
     
    <div
      class="el-form-item"
    >
      <!---->
      <div
        class="el-form-item__content"
        style="margin-left: 160px;"
      >
        <button
          class="el-button el-button--default"
          type="button"
        >
          <!---->
          <!---->
          <span>
            
        Huỷ
      
          </span>
        </button>
         
        <button
          class="el-button el-button--primary"
          type="button"
        >
          <!---->
          <!---->
          <span>
            
        Tạo
      
          </span>
        </button>
        <transition-stub
          name="el-zoom-in-top"
        >
          <!---->
        </transition-stub>
      </div>
    </div>
  </form>
</div>
`;
